array.go

```diff
@@ -1272,6 +1273,57 @@ func (a *Array) ConsolidateMetadataWithKey(encryptionType EncryptionType, key st
 	return nil
 }
 
+// GetTimeFromTimestamp returns a time.Time object for a time related TileDB datatype
+func GetTimeFromTimestamp(datatype Datatype, timestamp uint64) time.Time {
+	var then time.Time
+	switch datatype {
+	case TILEDB_DATETIME_YEAR:
+		then = time.Date(int(timestamp), 1, 1, 0, 0, 0, 0, time.UTC)
+	case TILEDB_DATETIME_MONTH:
+		now := time.Now()
+		then = time.Date(now.Year(), time.Month(timestamp), 1, 0, 0, 0, 0, time.UTC)
```

Comment:
The datetimes in tiledb are deltas from unix epoch with a resolution of the specified time. So month isn't just 1-12. You need to convert the number of months since epoch into the datetime. Same of all datetime types.

Reply:
Changed the logic